core/system/inc/linker.h

```diff
@@ -68,6 +70,9 @@ typedef struct {
     uint32_t *flash_end;
     uint32_t *sram_start;
     uint32_t *sram_end;
+
+    /* Privileged system IRQ hooks */
+    const UvisorPrivSystemIRQHooks * const priv_sys_irq_hooks;
```

Comment:
If here you want the ptr to the struct, then the macro to set the hooks should declare a sturct variable **and** declare a pointer-to-struct variable that contains the address of that struct.

The pointer would then be what here is called `priv_sys_irq_hooks`.